java/src/main/java/ai/rapids/cudf/JCudfSerialization.java

```diff
@@ -241,17 +380,28 @@ public void copyBytesToArray(byte[] dest, int destOffset, BufferType srcType, lo
    * Visible for testing
    */
   static class ColumnProvider extends ColumnBufferProvider {
-    private final HostColumnVector column;
+    private final ColumnViewAccess<HostMemoryBuffer> column;
     private final boolean closeAtEnd;
+    private final ColumnBufferProvider[] childProviders;
 
-    ColumnProvider(HostColumnVector column, boolean closeAtEnd) {
+    ColumnProvider(ColumnViewAccess<HostMemoryBuffer> column, boolean closeAtEnd) {
       this.column = column;
       this.closeAtEnd = closeAtEnd;
+      if (getType().isNestedType()) {
+        int numChildren = column.getNumChildren();
+        childProviders = new ColumnBufferProvider[numChildren];
+        for (int i = 0; i < numChildren; i++) {
+          // TODO: Is close parameter correct here?
```

Comment:
Yes it is for host columns, but not for device columns.  Which is another reason to separate ColumnViewAccess into two classes.  They are semantically different.